packages/react/src/components/DateTimePicker/DateTimePicker.jsx

```diff
@@ -682,6 +686,7 @@ const DateTimePicker = ({
           className={classnames(`${iotPrefix}--date-time-picker__menu`, {
             [`${iotPrefix}--date-time-picker__menu-expanded`]: isExpanded,
           })}
+          style={{ ...style, '--zIndex': zIndex ?? undefined }}
```

Comment:
The zIndex prop will already be undefined if not provided. We should either do a check that creates the style object with or without the zIndex or we should just have the alternative value be 0 which is the default you have set in the sass file. This same comment applies to the other DateTimePicker.